File: app/src/main/java/com/example/architechturestartercode/data/movies/MoviesRepository.java
package com.example.architechturestartercode.data.movies;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.example.architechturestartercode.data.movies.datasource.movies.local.MoviesLocalDataSource;
import com.example.architechturestartercode.data.movies.datasource.movies.remote.MoviesNetworkResponse;
import com.example.architechturestartercode.data.movies.datasource.movies.remote.MoviesRemoteDataSource;
import com.example.architechturestartercode.data.movies.model.Movie;

import java.util.List;

public class MoviesRepository {
    MoviesLocalDataSource moviesLocalDataSource;
    MoviesRemoteDataSource moviesRemoteDataSource;

    public MoviesRepository(Application application) {
        this.moviesLocalDataSource = new MoviesLocalDataSource(application);
        this.moviesRemoteDataSource = new MoviesRemoteDataSource();
    }

    public  void getAllMovies(MoviesNetworkResponse response) {
        moviesRemoteDataSource.getMovies(response);
    }

    public void addToFav(Movie movie) {
        moviesLocalDataSource.insertMovie(movie);
    }

    public void deleteMovie(Movie movie) {
        moviesLocalDataSource.deleteMovie(movie);
    }

    LiveData<List<Movie>> getFavMovies() {
        return moviesLocalDataSource.getMovies();
    }
}

File: app/src/main/java/com/example/architechturestartercode/data/movie/MoviesRepository.kt
package com.example.architechturestartercode.data.movie

import androidx.lifecycle.LiveData
import com.example.architechturestartercode.data.movie.datasource.local.MoviesLocalDataSource
import com.example.architechturestartercode.data.movie.datasource.remote.MoviesRemoteDataSource
import com.example.architechturestartercode.data.movie.model.Movie

class MoviesRepository(
    private val remoteDataSource: MoviesRemoteDataSource,
    private val localDataSource: MoviesLocalDataSource,
) {


    suspend fun getAllMovies(): Result<List<Movie>> {
        return remoteDataSource.getAllMovies()
    }

    suspend fun insertMovieToFav(movie: Movie) {
        localDataSource.insertMovie(movie)
    }

    suspend fun deleteMovieFromFav(movie: Movie) {
        localDataSource.deleteMovie(movie)
    }

    fun getAllFavMovies(): LiveData<List<Movie>> {
        return localDataSource.getAllMovies()
    }
}

